### PR TITLE
feat: kosko update

### PR DESCRIPTION
**Pull request by revolunet**
- embed `kosko/env/{merge,paths}` locally instead of private internal APIs
 - replace `Environment` calls with `createNodeCJSEnvironment` in tests

### CODE REVIEW

**Comment by revolunet** [comment]
@douglasduteil need your help for the typescript fixes /o\

**Comment by douglasduteil** [comment]
> embed kosko/env/{merge,paths} locally instead of private internal APIs 

Are we rewriting the existing `kosko/env/{merge,paths}` of mirroring it ?

File: src/utils/merge.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+export function merge(data: any[]): any {
+  return deepMerge.all(data, {
+    isMergeableObject: isPlainObject,
+  });
+}
+
+export async function mergeAsync(data: any[]): Promise<any> {
+  const values = await Promise.all(data);
+  return merge(values);
+}
```
Comment (douglasduteil) [inline src/utils/merge.ts:16]: Shouldn't we run the actual code instead of copy/paste it ?

File: src/utils/paths.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+export interface Paths {
+  global: string;
+  component: string;
+}
+
+export function formatPath(path: string, data: Record<string, string>): string {
+  return path.replace(rTemplate, (s, key) => {
+    return data[key] || s;
+  });
+}
```
Comment (douglasduteil) [inline src/utils/paths.ts:14]: Shouldn't we run the actual code instead of copy/paste it ?

**Review by douglasduteil — COMMENTED** [review]
Did you find copy/paste of kosko internal code better that having a proxy of it ?

**Comment by douglasduteil** [comment]
Similar to https://github.com/SocialGouv/kosko-charts/pull/441 ?

**Pull request closed without merging**